eth2-ssv/ibft/commit.go

```diff
@@ -0,0 +1,112 @@
+package ibft
+
+import (
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"github.com/bloxapp/ssv/ibft/pipeline/auth"
+
+	"go.uber.org/zap"
+
+	"github.com/bloxapp/ssv/ibft/pipeline"
+	"github.com/bloxapp/ssv/ibft/proto"
+)
+
+func (i *Instance) commitMsgPipeline() pipeline.Pipeline {
+	return pipeline.Combine(
+		auth.MsgTypeCheck(proto.RoundState_Commit),
+		auth.ValidateLambdas(i.State.Lambda),
+		auth.ValidateRound(i.State.Round),
+		auth.ValidatePKs(i.State.ValidatorPk),
+		auth.ValidateSequenceNumber(i.State.SeqNumber),
+		auth.AuthorizeMsg(i.Params),
+		i.uponCommitMsg(),
+	)
+}
+
+// CommittedAggregatedMsg returns a signed message for the state's committed value with the max known signatures
+func (i *Instance) CommittedAggregatedMsg() (*proto.SignedMessage, error) {
+	if i.State.PreparedValue == nil {
+		return nil, errors.New("state not prepared")
+	}
+
+	msgs := i.CommitMessages.ReadOnlyMessagesByRound(i.State.Round)
+	if len(msgs) == 0 {
+		return nil, errors.New("no commit msgs")
+	}
+
+	var ret *proto.SignedMessage
+	var err error
+	for _, msg := range msgs {
+		if !bytes.Equal(msg.Message.Value, i.State.PreparedValue) {
+			continue
+		}
+		if ret == nil {
+			ret, err = msg.DeepCopy()
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			if err := ret.Aggregate(msg); err != nil {
+				return nil, err
+			}
+		}
+	}
+	return ret, nil
+}
+
+func (i *Instance) commitQuorum(round uint64, inputValue []byte) (quorum bool, t int, n int) {
+	// TODO - calculate quorum one way (for prepare, commit, change round and decided) and refactor
+	cnt := 0
+	msgs := i.CommitMessages.ReadOnlyMessagesByRound(round)
+	for _, v := range msgs {
+		if bytes.Equal(inputValue, v.Message.Value) {
+			cnt++
+		}
+	}
+	quorum = cnt*3 >= i.Params.CommitteeSize()*2
```

Comment:
change this to a general hasQuorum helper that just takes in numbers and is able to return a boolean as it was used in a few places